tensorflow_addons/seq2seq/beam_search_decoder.py

```diff
@@ -225,6 +224,15 @@ def _check_batch_beam(t, batch_size, beam_width):
     return tf.Assert(condition, [error_message])
 
 
+def _as_shape(shape):
```

Comment:
the input is not a strict shape, it can be a tensor as well, probably rename it to not be shape, which is bit confusing.

Reply:
Maybe something like `dims_or_shape`?

Reply:
shouldn't it be shape_or_tensor?

Reply:
It could also be a Python list of dimensions.

Reply:
Hum, since it is polymorphic, how about just call it input?

Reply:
Ok, I went with `value` as `input` clashes with a Python builtin.